SOLIDHomework.Core/Services/InventoryService.cs

```diff
@@ -1,20 +1,20 @@
-﻿using SOLIDHomework.Core.Model;
+﻿using SOLIDHomework.Core.Interfaces;
+using SOLIDHomework.Core.Model;
 using System;
 
 namespace SOLIDHomework.Core.Services
 {
-    public class InventoryService
+    public class InventoryService : IInventoryService
     {
-        // that is Database-based service 
+
         public void ReserveInventory(ShoppingCart cart)
         {
             foreach (OrderItem item in cart.OrderItems)
             {
                 try
                 {
                     InventoryService inventoryService = new InventoryService();
```

Comment:
Why do you create an instance of service inside the service?

Reply:
take out InventoryService invocation.